openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/parameter/ParameterGenerator.java

```diff
@@ -20,13 +20,26 @@
 
 import io.ballerina.compiler.syntax.tree.ParameterNode;
 import io.ballerina.openapi.core.generators.client.diagnostic.ClientDiagnostic;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.Parameter;
 
 import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
 
 public interface ParameterGenerator {
     //type handler attribute
     Optional<ParameterNode> generateParameterNode();
     List<ClientDiagnostic> getDiagnostics();
 
+    default Schema getSchemaWithDetails(Parameter parameter) {
```

Comment:
Any possibility on updating the method signature ? I think this would be more cleaner.
```suggestion
    default Optional<Schema> getSchemaWithDetails(Parameter parameter) {
```

Reply:
All the parameters should have a schema. The null check for the schema is to avoid the NPE when an invalid OpenAPI specification is provided. So having an optional schema does not make sense here